### PR TITLE
Add new choice logic and unittests

### PR DESCRIPTION
**Pull request by lindwan3**
If config1 has a different case than config2, in edit-config remove deletion of case that is no longer being used.

Based on this:
Query regarding rfc6241 - 7.2 , same conceptual node and choice statement
https://mailarchive.ietf.org/arch/browse/netconf/

### CODE REVIEW

File: src/ncdiff/netconf.py [hunk 1]
```diff
@@ -983,18 +984,31 @@ def node_sub(self, node_self, node_other, depth=0):
...
+            s_node = self.device.get_schema_node(child_other)
             if siblings:
                 siblings[-1].addnext(child_self)
             else:
-                node_self.append(child_self)
-            s_node = self.device.get_schema_node(child_other)
+                # Append node if:
+                # Node not in case
+                # Node in case but choice node not in self
+                # Node in case and choice not in self but case also in self
```
Comment (yuekyang) [inline src/ncdiff/netconf.py:1004]: How about `Node in case and choice node in self and the same case also in self`?

File: src/ncdiff/netconf.py [hunk 2]
```diff
@@ -983,18 +984,31 @@ def node_sub(self, node_self, node_other, depth=0):
...
+                if s_node.getparent().get('type') == 'case':
+                    choice_node = s_node.getparent().getparent()
+                    if choice_node not in choice_nodes or \
+                       s_node.getparent() == choice_nodes[choice_node]:
```
Comment (yuekyang) [inline src/ncdiff/netconf.py:1008]: `choice_node in choice_nodes and s_node.getparent() == choice_nodes[choice_node]:`

**Review by yuekyang — APPROVED** [review]

File: src/ncdiff/netconf.py (hunk 1 above)
Comment (lindwan3) [reply src/ncdiff/netconf.py:1004]: I did a unit test for this
https://github.com/CiscoTestAutomation/ncdiff/pull/27/files#diff-a32bac34fbf81e957a4ced64cf60128f21ff9ea8d403fcc89b3b39906a0af8c0R1139

File: src/ncdiff/netconf.py (hunk 2 above)
Comment (lindwan3) [reply src/ncdiff/netconf.py:1008]: Is "choice_node in choice_nodes and" necessary? If the logic reaches "s_node.getparent() == choice_nodes[choice_node]" we already know "choice_node in choice_nodes"

File: src/ncdiff/netconf.py (hunk 1 above)
Comment (yuekyang) [reply src/ncdiff/netconf.py:1004]: You are right. No change is needed.

File: src/ncdiff/netconf.py (hunk 2 above)
Comment (yuekyang) [reply src/ncdiff/netconf.py:1008]: You are right. No change is needed.

**Pull request merged**